sdk/trace/src/main/java/io/opentelemetry/sdk/trace/TracerSharedState.java

```diff
@@ -101,19 +105,24 @@ void addSpanProcessor(SpanProcessor spanProcessor) {
    * @return {@code true} if tracing is stopped.
    */
   boolean isStopped() {
-    return isStopped;
+    synchronized (lock) {
+      return isStopped != null && isStopped.isSuccess();
+    }
   }
 
   /**
    * Stops tracing, including shutting down processors and set to {@code true} {@link #isStopped()}.
+   *
+   * @return a {@link CompletableResultCode} that will be completed when the span processor is shut
+   *     down.
    */
-  void stop() {
+  CompletableResultCode stop() {
```

Comment:
Maybe it's worth renaming this to `shutdown` for consistency (even though it's an internal method)